### PR TITLE
Add setting conf for Resettable Apps

### PR DESCRIPTION
**Pull request by mike-sul**
- Added a new param `--restorable-apps`
- If not specified then Restorable Apps are OFF
- If specified just `--restorable-apps` with no value then turn Restorable Apps ON and use `compose_apps` as an App list
- If `--restorable-apps` is specified and defined with a value then a list of restorable Apps == UNION(reset_apps, compose_apps)

Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -608,6 +616,16 @@ int main(int argc, char **argv)
...
 		device.put("overrides.pacman.compose_apps", "\"" + options.apps + "\"");
 	}
+	string reset_apps_root = options.sota_config_dir + "/reset-apps";
+	device.put("overrides.pacman.reset_apps_root", "\"" + reset_apps_root + "\"");
+
+	if (!options.reset_apps.empty()) {
+		device.put("overrides.pacman.reset_apps", "\"" + options.reset_apps + "\"");
+	} else if (boost::filesystem::exists(reset_apps_root)) {
+		// if `reset-apps` is not specified but a system image is preloaded with Restorable/Resettable Apps then force reset-apps ON
+		cout << "Device is preloaded with Resettable Apps, turn their usage ON" << endl;
```
Comment (doanac) [inline src/main.cpp:626]:
```suggestion
		cout << "Device is preloaded with Resettable Apps, turning their usage ON" << endl;
```

File: src/main.cpp [hunk 2]
```diff
@@ -111,7 +112,14 @@ static bool _get_options(int argc, char **argv, Options &options)
...
 		("apps,a", po::value<string>(&options.apps),
-		 "Configure package-manager for this comma separate list of apps.")
+		"Configure package-manager for this comma separate list of apps.")
+		// --restorable-apps : turn ON Restorable Apps usage, its list == compose_apps or all Target apps
+		// --restorable-apps "app-01[,app-02]" : turn ON Restorable Apps usage, its list == UNION(compose_apps, app-01[,app-02])
+		// if --restorable-apps is not specified or `--restorable-apps ""` then it means turning restorable Apps off
+		 ("restorable-apps,A", po::value<string>(&options.restorable_apps)->implicit_value(" "),
+		 "Configure package-manager for this comma separate list of Restorable apps."
+		 "It's set to an empty list automatically if not specified and a system image is preloaded with Restorable Apps."
+		 "Restorable App list = UNION(compose-apps, restorable-apps),")
```
Comment (doanac) [inline src/main.cpp:122]: I don't want to nit, but I've read this sentence a few times and the "if not specified" part keeps making me think you can't turn off restorable apps. i.e it defaults to all. Then I read your code comments and it becomes clear and expected. I think you should incorporate the comments into the usage message. They are more clear.

Comment (mike-sul) [reply src/main.cpp:122]: Makes sense

Comment (mike-sul) [reply src/main.cpp]: updated

**Pull request merged**